### ISSUE TITLE
Run upgrade jobs in parallel instead of in sequence

### ISSUE THREAD

**Issue by ihmccreery** [kubernetes/kubernetes#22104]
This is a pretty hazy issue right now, and I'll scope it out more later, but for now, the plan:

For each `X`-`Y` version skew, run 4 suites:

- Deploy at `X` and run Kubectl tests at `Y`
- Deploy at `X`, upgrade master to `Y`, and run tests at `X`
- Deploy at `X`, upgrade master and nodes to `Y`, and run tests at `X`
- Deploy at `X`, upgrade master and nodes to `Y`, and run tests at `Y`

This should give us the same coverage as we have now, but without chaining runs together.  I need this refactoring because I'll have to fetch and unpack different tests between cluster-up and test.

**Comment by ihmccreery** [comment]
cc @kubernetes/sig-testing

**Comment by ihmccreery** [comment]
The blocker here is being able to run two `ginkgo test` runs in the same Jenkins job and deal with the JUnit output in a sane way.

**Comment by ihmccreery** [comment]
Okay, after talking a bit with @spxtr, we're going to have the following 6 jobs running for each version skew:

- Deploy at `X` and run Kubectl tests at `Y`
- Deploy at `X`, upgrade master to `Y`, and run tests at `X`
- Deploy at `X`, upgrade master and nodes to `Y`, and run tests at `X`
- Deploy at `X`, upgrade entire cluster to `Y`, and run tests at `Y`
- Deploy at `X`, and run upgrade test across upgrading master to `Y`
- Deploy at `X`, and run upgrade test across upgrading entire cluster to `Y`

**Comment by ihmccreery** [comment]
This should be something we can implement entirely in `e2e.go`, using `SynchronizeBeforeSuite` and a flag.  Will write it up on Monday.

**Comment by ihmccreery** [comment]
With https://github.com/kubernetes/kubernetes/pull/24428 and friends in, this is done.  They're not stable, but that's a separate issue.
